crates/ide_completion/src/render.rs

```diff
@@ -955,6 +1046,10 @@ fn main() {
                         ),
                         lookup: "foo",
                         detail: "-> ()",
+                        relevance: CompletionRelevance {
+                            exact_name_match: false,
+                            exact_type_match: true,
```

Comment:
This is a false positive on `exact_type_match`, I need to dig into this.

Reply:
https://github.com/rust-analyzer/rust-analyzer/blob/be7a31fbd64943f71afe11b0413c99496526dddc/crates/ide_completion/src/context.rs#L293-L296

This false positive is caused by `expected_type` being incorrect. I've confirmed if I update this test to `foo(any_prefix$0)` the `exact_type_match` here is set to false (as it should be). 

I'm going to put this PR on pause while I try to go improve the underlying issue, rather than implementing another workaround here. I did try for a bit to figure out a way to do this like a week ago and didn't have any luck, so hopefully this time I can figure something out :laughing:

Reply:
I've proposed a fix for `expected_type` in #8008.

Reply:
Fixed in #8008